scripts/ember_customization.py

```diff
@@ -22,6 +31,84 @@ def apply_custom_ramping(n):
             n.links.loc[idx, param] = param_value
 
 
+def apply_2023_nuclear_decommissioning(n, year=2023):
+    if year == 2023:
+        nuclear_info = {
+            "Isar 2": {"coords": [12.29315, 48.60560556], "country": 'DE', "dateout": "2023-04-15"},
+            "Emsland": {"coords": [7.317858333, 52.47423056], "country": 'DE', "dateout": "2023-04-15"},
+            "Neckarwestheim 2": {"coords": [9.175, 49.04111111], "country": 'DE', "dateout": "2023-04-15"},
+        }
+
+    seen_plants = []
+    for plant, info in nuclear_info.items():
+        # network details
+        country = nuclear_info[plant]["country"]
+        country_buses = n.buses.query("country in @country").index
+        network_coords = (
+            n.buses.loc[n.links.query("carrier == 'nuclear' and bus1 in @country_buses").bus1, ["x", "y"]]
+            .set_index(n.links.query("carrier == 'nuclear' and bus1 in @country_buses").index)
+        )
+        network_coords = network_coords.drop(index=seen_plants)
+
+        # plant details
+        dateout = pd.Timestamp(info["dateout"])
+        px, py = info["coords"]
+        dx = network_coords["x"].to_numpy() - px
+        dy = network_coords["y"].to_numpy() - py
+
+        # closest country nuclear plant
+        dist = np.sqrt(dx**2 + dy**2)
+        nearest_gen = network_coords.index[np.argmin(dist)]
+
+        # decommission
+        n.links_t.p_max_pu[nearest_gen] = (
+            n.links.loc[nearest_gen].p_max_pu * ((n.snapshots < dateout).astype(int))
+        )
+
+        seen_plants.append(nearest_gen)
+# Additions to ember_customization.py
+
+
+def apply_hourly_gas_prices(n, config):
+    """
+    Apply hourly gas prices to gas-fired generators (OCGT and CCGT).
+    Sets time-varying marginal costs based on hourly gas spot prices.
+    Adjusts static marginal cost to VOM only.
+    """
+    
+    repo_root = os.path.dirname(os.path.abspath(__file__)) 
+    csv_path = os.path.join(repo_root, "validation", "ember_data", "hourly_fuel_costs.csv")
+    
+    df = pd.read_csv(csv_path)
+    df['timestamp'] = pd.to_datetime(df['timestamp'])
+    df.set_index('timestamp', inplace=True)
+    
+    if not (df.index.equals(n.snapshots)):
+        logger.warning("Snapshot indices do not match exactly. Ensure timestamps align with network snapshots.")
+       
+    
+    avg_fuel = config['costs']['fuel']['gas']
+    price_col = 'GAS_SPOT_PRICE_EUR_PER_MWH'
+    prices = df[price_col]
+    
+    carriers = ['gas']
```

Comment:
I'd suggest having carriers as function input, and then part of ember_settings in the config file which is then passed through - the carriers is then a set, either empty or containing 'gas', 'coal', and/or 'lignite' (error if it's sth else).

then the price_col would be a dictionary with

{"gas": "GAS_SPOT_PRICE_EUR_PER_MWH", "coal": "COAL_SPOT_PRICE_EUR_PER_MWH"}

and for lignite, you'd create a heuristic (per_unit as discussed in our call) and then generate it yourself from the static value.